From e1de3374179bbac5bc6416ba32f35f339112b422 Mon Sep 17 00:00:00 2001
From: Ben Myers <bpm@sgi.com>
Date: Tue, 10 Dec 2013 14:59:31 -0600
Subject: [PATCH] xfs: fix calculation of freed inode cluster blocks
Git-commit: 8e825e3a02ff20973154559c33e662cacedc4458
Patch-mainline: v3.14-rc1

rec.ir_startino is an agino rather than an ino.  Use the correct macro
when dealing with it in xfs_difree.

Signed-off-by: Ben Myers <bpm@sgi.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>

Signed-off-by: Jan Kara <jack@suse.cz>

---
 fs/xfs/xfs_ialloc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/fs/xfs/xfs_ialloc.c b/fs/xfs/xfs_ialloc.c
index f9e6d74..a799ce3 100644
--- a/fs/xfs/xfs_ialloc.c
+++ b/fs/xfs/xfs_ialloc.c
@@ -1230,7 +1230,7 @@ xfs_difree(
 		}
 
 		xfs_bmap_add_free(XFS_AGB_TO_FSB(mp,
-				agno, XFS_INO_TO_AGBNO(mp,rec.ir_startino)),
+				agno, XFS_AGINO_TO_AGBNO(mp, rec.ir_startino)),
 				XFS_IALLOC_BLOCKS(mp), flist, mp);
 	} else {
 		*delete = 0;
-- 
1.6.0.2

